components/atom/videoPlayer/src/hooks/useVideoPlayer.js

```diff
@@ -0,0 +1,35 @@
+import {VIMEO, YOUTUBE, BASE_CLASS} from '../settings.js'
+import {VimeoPlayer, YouTubePlayer} from '../components/index.js'
+
+const matcher = (arrayPattern, value) =>
+  arrayPattern.find(pattern => value.includes(pattern))
+
+const useVideoPlayer = ({
+  as: As = 'div',
+  children = 'Not supported media type',
+  className,
+  src = '',
+  ...props
+} = {}) => {
+  if (matcher([YOUTUBE.VIDEO_TYPE, YOUTUBE.SHORT_URL], src)) {
+    return [
+      YouTubePlayer,
+      {
+        className: [BASE_CLASS, `${BASE_CLASS}-youtubePlayer`].join(' '),
+        src,
+        ...props
+      }
+    ]
+  } else if (matcher([VIMEO.VIDEO_TYPE], src)) {
+    return [
+      VimeoPlayer,
+      {
+        className: [BASE_CLASS, `${BASE_CLASS}-vimeoPlayer`].join(' '),
+        src,
+        ...props
+      }
+    ]
+  }
+  return ['h1', {className: BASE_CLASS, children, ...props}]
+}
```

Comment:
Is this an agreement of how to create these kind of structures in Sui?

Probably I'm missing some kind of benefit or architectural pattern, but IMO this is making the same effect than the already existing code, but in a less obvious way, delegating a some representation responsibilities to the `useVideoPlayer` hook.

All of this is still entirely my opinion, but without knowing the benefits of applying this alternative approach, it's more difficult in my opinion to read the code and follow the logic flow. For example is less obvious were the `vimeoPlayer` style class is used. In the already  existing approach, the Vimeo player component self-contains both all the needed logic and css class references, and the switch mechanism has a very specific and limited responsibility.

Maybe we can have a discussion about this when you have a moment for us!

Reply:
reduces the VDOM tree reconciliation. Take a look on react-dev-tools, there is non `Switcher` component